### PR TITLE
Fast AWS-native disk resize

### PR DESCRIPTION
**Pull request by bgandon**
Hi there,

In #122, we've seen that when resizing disks, the `optimizing` state is only a backend concern on the AWS EBS servers infrastructure, for delivering the promised performance. So for a Bosh CPI concern, there is no need to wait for the EBS backend reorganisation to finish with the `completed` state. This is good news as our production clusters were taking days for the optimisations to finish.

Here in this PR, we provide an implementation for the CPI to move on once the [VolumeModification](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_VolumeModification.html) reaches the `optimizing` or `completed` states.

With this new approach, the `aws.extend_ebs_volume.wait_time_factor` config property introduced in #119 is no more necessary, so it has been removed. Which is also good news because its concept was a bit sloppy, as mentioned during review.

Unit tests are passing and we've tested successfully this code for resizing some development cluster. It' now lightning fast!

We need quick reviews on this code, in order to resize big data clusters in the upcoming days.
Thank you!

### CODE REVIEW

**Review by jpalermo — APPROVED** [review]
👍

File: src/bosh_aws_cpi/lib/cloud/aws/resource_wait.rb [hunk 1]
```diff
@@ -74,22 +74,16 @@ def self.for_volume(args)
...
       volume_modification = args.fetch(:volume_modification) { raise ArgumentError, 'volume_modification object required' }
-      target_state = args.fetch(:state) { raise ArgumentError, 'state symbol required' }
+      target_states = args.fetch(:states) { raise ArgumentError, 'states array required' }
       valid_states = ['modifying', 'optimizing', 'completed', 'failed']
-      validate_states(valid_states, target_state)
-      wait_time_factor = args.fetch(:wait_time_factor, 1).to_i
+      target_states.each { |target_state| validate_states(valid_states, target_state) }
+      target_state_desc = target_states.join(' or ')
 
-      ignored_errors = []
```
Comment (beyhan) [inline src/bosh_aws_cpi/lib/cloud/aws/resource_wait.rb:82]: Why we don't need to ignore these errors any more? I couldn't follow this and doesn't seem to be related to this change.

**Review by beyhan — COMMENTED** [review]
In general the removal of `aws.extend_ebs_volume.wait_time_factor` is a backward incompatible change. It is most probably OK to remove it because it was added with the last release. Is it sure that this isn't required any more?

File: src/bosh_aws_cpi/lib/cloud/aws/resource_wait.rb (hunk 1 above)
Comment (bgandon) [reply src/bosh_aws_cpi/lib/cloud/aws/resource_wait.rb:82]: I bumped into this while reading my own copy/paste-based code. 😅

I figured that in the original code I've copied, we have to ignore "disk resource missing" errors while creating a new disk, because before it is actually created, some API calls may error in this way.

But when resizing a disk, then the API resource for the disk should definitely exist. So we don't ignore such errors.

**Comment by bgandon** [comment]
> _Is it sure that this isn't required any more?_

Yes, definitely. When not waiting for the `optimizing` phase to finish, the `VolumeModification` operation is lightning fast, probably due to a kind of “thin provisioning” strategy on EBS side. So, we don't need any extra waiting time anymore. We need to properly timeout when the `VolumeModification` API call takes way longer than it should. Just as we do with other API calls.

**Review by beyhan — APPROVED** [review]

**Comment by bgandon** [comment]
We have two approvals, let's merge shall we?
